docker/Dockerfile.git

```diff
@@ -0,0 +1,25 @@
+FROM avsm/docker-opam:ubuntu-trusty-4.02.1
+MAINTAINER Shrutarshi Basu <basus@cs.cornell.edu>
+RUN sudo apt-get -y update
+RUN sudo apt-get -y upgrade
+RUN sudo apt-get -y install zlib1g-dev
+RUN cd ~/opam-repository && git pull && opam update -u -y && opam upgrade -y
+RUN opam install core async async_extended fieldslib cmdliner cstruct sexplib \
+                 ulex ipaddr yojson base64 cohttp quickcheck ounit pa_ounit \
+                 tcpip ocamlgraph
+RUN eval `opam config env`
```

Comment:
I think you can replace the apt-get lines above with `opam depext -i core <...>`.  This will install all the external dependencies required by the packages.

You might also want to use the [OPAM 1.2 pinning workflow](https://opam.ocaml.org/blog/opam-1-2-pin/).  TL;DR: add an `opam` file to this repository, and then just do `opam pin add <dir>` to install all the development versions that you have below.  This will make all the `make install` lines redundant, as OPAM will just install the development versions.